### ISSUE TITLE
APM is unable to install with scoped packages

### ISSUE THREAD

**Issue by TimvdLippe** [atom/apm#702]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

We are using the build script provided by Atom (https://raw.githubusercontent.com/atom/ci/master/build-package.sh) which should be installing our packages. Since our Atom package is built with Typescript, we have to first compile the code before we can start the actual tests. This requires installation of our packages with NPM/Yarn.

If APM is then invoked to install packages, it gets into an endless loop trying to remove and install scoped packages: https://travis-ci.org/Polymer/atom-plugin/builds/211494054#L234

### Steps to Reproduce

1. Run `npm install` on an Atom package with a scoped package (for example `@types/atom`)
2. Use the build script
3. Run on Travis, but reproduces locally too

**Expected behavior:** Installation continues

**Actual behavior:** Endless loop trying to remove scoped packages during installation

**Reproduces how often:** 100%

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

```
Downloading latest Atom release...
Using Atom version:
Atom    : 1.16.0-beta0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
Using APM version:
apm  1.16.1
npm  3.10.5
node 6.9.5 x64
python 2.7.3
git 1.8.5.6
```

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by 43081j** [comment]
FYI this can also be reproduced IIRC by:

```
npm install
apm clean
```

the clean call gets stuck in loops trying to clean scoped packages.

**Comment by TimvdLippe** [comment]
@43081j good call, it indeed was the clean command. I submitted a fix in #705

**Comment by Alhadis** [comment]
I wish I could physically punch `apm`.
